website/static/js/logFeed.js

```diff
@@ -1,238 +1,245 @@
+        var size = limitLogs ? LOG_PAGE_SIZE_LIMITED : LOG_PAGE_SIZE;
+        var query = { 'page[size]': size, 'page': page, 'embed': ['original_node', 'user', 'linked_node', 'template_node']};
+        if (node.link) {
+            query.view_only = node.link;
         }
-        return ret;
-    });
-
-    //helper function to strip the slash for file or folder in log template
-    self.stripSlash = function(path){
-        return path.replace(/(^\/)|(\/$)/g, '');
-    };
-
-    //helper funtion to determine the type for removing in log template
-    self.pathType = function(path){
-        return path.match(/\/$/) ? 'folder' : 'file';
-    };
+        return $osf.apiV2Url(urlPrefix + '/' + node.id + '/logs/', { query: query});
+    }
 };
 
-/**
-  * View model for a log list.
-  * @param {Log[]} logs An array of Log model objects to render.
-  * @param url the url ajax request post to
-  */
-var LogsViewModel = oop.extend(Paginator, {
-    constructor: function(logs, url) {
-        this.super.constructor.call(this);
+var LogFeed = {
+
+    controller: function(options) {
         var self = this;
-        self.loading = ko.observable(false);
-        self.logs = ko.observableArray(logs);
-        self.url = url;
-        self.anonymousUserName = '<em>A user</em>';
-
-        self.tzname = ko.pureComputed(function() {
-            var logs = self.logs();
-            if (logs.length) {
-                var tz =  moment(logs[0].date.date).format('ZZ');
-                return tz;
+        self.node = options.node;
+        self.activityLogs = m.prop();
+        self.logRequestPending = m.prop();
+        self.limitLogs = options.limitLogs;
+        self.failed = false;
+        self.paginators = m.prop([]);
+        self.nextPage = m.prop();
+        self.prevPage = m.prop();
+        self.firstPage = m.prop();
+        self.lastPage = m.prop();
+        self.totalPages = m.prop();
+        self.currentPage = m.prop();
+        self.pageToGet = m.prop();
+
+        self.getLogs = function _getLogs (url) {
+            self.activityLogs([]); // Empty logs from other projects while load is happening;
+            function _processResults (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                });
+
+                self.activityLogs(result.data);  // Set activity log data
+                self.nextPage = result.links.next;
+                self.prevPage = result.links.prev;
+                self.firstPage = result.links.first;
+                self.lastPage = result.links.last;
+
+                var params = $osf.urlParams(url);
+                var page = params.page ? params.page : 1;
+                self.currentPage(parseInt(page));
+                self.totalPages = Math.ceil(result.links.meta.total / result.links.meta.per_page);
```

Comment:
Ah, actually ```totalPages``` doesn't need to be an m.prop, so I'll change it to have some default value (in which case the way it's assigned and invoked in the rest of the file will be correct).